apps/src/code-studio/components/ExportAllowedDialog.jsx

```diff
@@ -0,0 +1,744 @@
+      showSendToPhone,
+      exportError,
+      apkError,
+      apkUri = ''
+    } = this.state;
+    const waiting = exporting || generatingApk;
+    const error = exportError || apkError;
+    const {appType, channelId} = this.props;
+    const titleText = waiting
+      ? 'Creating Android Package...'
+      : error
+      ? 'Error creating Android Package'
+      : 'The Android Package was created successfully';
+    const headerText = waiting
+      ? 'Please wait for about <b>15 minutes</b>.'
+      : error || 'Send this link to your device to install the app.';
+    return (
+      <div>
+        <div style={styles.section}>
+          <p style={styles.title}>{titleText}</p>
+        </div>
+        <div style={styles.section}>
+          <p style={styles.p} dangerouslySetInnerHTML={{__html: headerText}} />
+        </div>
+        <div style={styles.section}>
+          {waiting && (
+            <i style={styles.spinner} className="fa fa-spinner fa-spin" />
+          )}
+          {!waiting && !error && (
+            <div>
+              <div style={styles.apkUriContainer}>
+                <input
+                  type="text"
+                  onClick={this.onInputSelect}
+                  readOnly="true"
+                  value={apkUri}
+                  style={styles.expoInput}
+                />
+              </div>
+              <button
+                type="button"
+                style={styles.sendToPhoneButton}
+                onClick={this.showSendToPhone}
+              >
+                <div style={styles.sendToPhoneButtonBody}>
+                  <i
+                    className="fa fa-mobile-phone"
+                    style={styles.sendToPhoneIcon}
+                  />
+                  <span>{i18n.sendToPhone()}</span>
+                </div>
+              </button>
+              {showSendToPhone && (
+                <SendToPhone
+                  channelId={channelId}
+                  appType={appType}
+                  downloadUrl={apkUri}
+                  isLegacyShare={false}
+                  styles={{label: styles.phoneLabel}}
+                />
+              )}
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  getActionButtonInfo() {
+    const {screen, exporting, generatingApk} = this.state;
+    const info = {
+      text: 'Next',
+      enabled: true
+    };
+    switch (screen) {
+      case 'generating':
+        info.text = 'Finish';
+        info.enabled = !exporting && !generatingApk;
+        break;
+      case 'export':
+        info.enabled = !exporting;
+        break;
+      case 'publish':
+        info.text = 'Create';
+        break;
+    }
+    return info;
+  }
+
+  render() {
+    const {
+      text: actionText,
+      enabled: actionEnabled
+    } = this.getActionButtonInfo();
+    const showShareWarning =
+      !this.props.canShareSocial &&
+      (this.props.appType === 'applab' || this.props.appType === 'gamelab');
```

Comment:
is there a specific reason why `weblab` is included in the "sharing enabled" logic, but not here?

Reply:
Good question. This copied the logic from `ShareAllowedDialog` - out of caution, I suggested to @ryansloan that we stick with the same logic for exporting. But I didn't try to revisit that logic.

Reply:
That said, I should probably just remove the `weblab` references since we don't plan to support exporting weblab sites as mobile apps.